kratos/geometries/brep_surface.h

```diff
@@ -549,6 +554,42 @@ class BrepSurface
         return GeometryData::KratosGeometryType::Kratos_Brep_Surface;
     }
 
+    /**
+     * @brief Set the Surrogate Outer Loop Geometries object
+     * @param pSurrogateOuterLoopArray 
+     */
+    void SetSurrogateOuterLoopGeometries(GeometrySurrogateArrayType pSurrogateOuterLoopArray)
```

Comment:
As far as I see the type is not a pointer so you are creating a copy each time you set it. You either pass it by reference or use a pointer.